packages/design-picker/src/components/style-variation-badges/index.tsx

```diff
@@ -10,23 +10,42 @@ interface BadgesProps {
 	variations: StyleVariation[];
 	onMoreClick?: () => void;
 	onClick?: ( variation: StyleVariation ) => void;
+	selectedVariation?: StyleVariation;
+	firstVariationToShow?: StyleVariation;
 }
 
 const Badges: React.FC< BadgesProps > = ( {
 	maxVariationsToShow = 4,
 	variations = [],
 	onMoreClick,
 	onClick,
+	selectedVariation,
+	firstVariationToShow,
 } ) => {
-	const variationsToShow = useMemo(
-		() => variations.slice( 0, maxVariationsToShow ),
-		[ variations, maxVariationsToShow ]
-	);
+	const variationsToShow = useMemo( () => {
+		return [
+			...( firstVariationToShow ? [ firstVariationToShow ] : [] ),
+			...( firstVariationToShow
+				? variations
+						.filter( ( variation ) => variation.slug !== firstVariationToShow.slug )
+						.slice( 0, maxVariationsToShow - 1 )
+				: [] ),
+			...( ! firstVariationToShow ? variations.slice( 0, maxVariationsToShow ) : [] ),
+		];
+	}, [ variations, maxVariationsToShow, firstVariationToShow ] );
```

Comment:
> it makes sense to keep the logic in the `Badges`.

Yeah, I think so. The logic in `UnifiedDesignPicker` is already complex enough that I want to avoid making it even more complex if possible. And this logic fits very well in what `Badges` does.

> `firstVariationToShow`, which is a bit of a mouthful

I renamed it to `firstVariation` so it's simpler 🙂 

> Also how do you feel about making this more readable by separating the two cases?

Good suggestion! Done in the last commit.